### ISSUE TITLE
issue about `mean_dice_val` calculation in unetr_btcv_segmentation_3d.ipynb

### ISSUE THREAD

**Issue by yiyixuxu** [Project-MONAI/tutorials#665]
see below code I took from the UNETR BTCV tutorial , since `dice_metric` is a cumulative metric, with `dice = dice_metric.aggregate().item()`, you already get the accumulated dice mean for all the samples evaluated - Is there any reason to take the average again in the end of validation step ( i.e.   `mean_dice_val = np.mean(dice_vals)`)?

Thanks!

```python
def validation(epoch_iterator_val):
    model.eval()
    dice_vals = list()
    with torch.no_grad():
        for step, batch in enumerate(epoch_iterator_val):
            val_inputs, val_labels = (batch["image"].cuda(), batch["label"].cuda())
            val_outputs = sliding_window_inference(val_inputs, (96, 96, 96), 4, model)
            val_labels_list = decollate_batch(val_labels)
            val_labels_convert = [
                post_label(val_label_tensor) for val_label_tensor in val_labels_list
            ]
            val_outputs_list = decollate_batch(val_outputs)
            val_output_convert = [
                post_pred(val_pred_tensor) for val_pred_tensor in val_outputs_list
            ]
            dice_metric(y_pred=val_output_convert, y=val_labels_convert)
            dice = dice_metric.aggregate().item()
            dice_vals.append(dice)
            epoch_iterator_val.set_description(
                "Validate (%d / %d Steps) (dice=%2.5f)" % (global_step, 10.0, dice)
            )
        dice_metric.reset()
    mean_dice_val = np.mean(dice_vals)
    return mean_dice_val
```

**Comment by Nic-Ma** [comment]
Hi @ahatamiz ,

Could you please help take a look at this code question? Maybe we should slightly update the tutorial?

Thanks in advance.

**Comment by achen46** [comment]
Hi @yiyixuxu 

`dice_metric.aggregate()` is used to obtain the aggregate dice for 14 classes (organs + background) in this tutorial. Furthermore, we take the mean to report a single average dice. 

Thanks

**Comment by ahatamiz** [comment]
Hi @yiyixuxu 

`dice_metric.aggregate()` is used to obtain the aggregate dice for 14 classes (organs + background) in this tutorial. Furthermore, we take the mean to report a single average dice.

Thanks

**Comment by Nic-Ma** [comment]
Hi @yiyixuxu ,

I think that depends on the `reduction` method you set for the metric:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/meandice.py#L38

Thanks.

**Comment by yiyixuxu** [comment]
Thanks @Nic-Ma 

In the tutorial the reducer is set to be "mean" , so it returns a scalar average across all 14 classes and across all the samples stored in the buffer - from my understanding we do not need to average again but I'm not sure if I understand correctly

`dice_metric = DiceMetric(include_background=True, reduction="mean", get_not_nans=False)`

**Comment by Nic-Ma** [comment]
Hi @ahatamiz ,

Could you please help share more details?

Thanks in advance.

**Comment by ahatamiz** [comment]
@yiyixuxu, Yes.  `dice_metric.aggregate()` simply returns a tensor of size 14. We take the mean to report the average dice. 

I would recommend running the tutorial and printing the values.

**Comment by yiyixuxu** [comment]
Thanks @ahatamiz 

this is an example of what it print out,  at each mini-batch `dice_metric.aggregate()` generate a scalar that gets append to `dice_vals`... 

Training (5 / 25 Steps) (loss=4.16187):  21%|██        | 5/24 [00:13<00:31,  1.65s/it]
Validate (X / X Steps) (dice=X.X):   0%|          | 0/6 [00:00<?, ?it/s]
Validate (0 / 10 Steps) (dice=0.00319):   0%|          | 0/6 [00:03<?, ?it/s]
Validate (0 / 10 Steps) (dice=0.00319):  17%|█▋        | 1/6 [00:03<00:18,  3.71s/it]global step: 0 val_step: 0, 
 dice_metric.aggregate(): 0.00318509666249156 
 dice_vals: [0.00318509666249156] 

Validate (0 / 10 Steps) (dice=0.00248):  17%|█▋        | 1/6 [00:06<00:18,  3.71s/it]
Validate (0 / 10 Steps) (dice=0.00248):  33%|███▎      | 2/6 [00:06<00:12,  3.13s/it]global step: 0 val_step: 1, 
 dice_metric.aggregate(): 0.002484228229150176 
 dice_vals: [0.00318509666249156, 0.002484228229150176] 

Validate (0 / 10 Steps) (dice=0.00247):  33%|███▎      | 2/6 [00:08<00:12,  3.13s/it]
Validate (0 / 10 Steps) (dice=0.00247):  50%|█████     | 3/6 [00:08<00:07,  2.63s/it]global step: 0 val_step: 2, 
 dice_metric.aggregate(): 0.002474189503118396 
 dice_vals: [0.00318509666249156, 0.002484228229150176, 0.002474189503118396] 

Validate (0 / 10 Steps) (dice=0.00231):  50%|█████     | 3/6 [00:12<00:07,  2.63s/it]
Validate (0 / 10 Steps) (dice=0.00231):  67%|██████▋   | 4/6 [00:12<00:06,  3.37s/it]global step: 0 val_step: 3, 
 dice_metric.aggregate(): 0.0023066357243806124 
 dice_vals: [0.00318509666249156, 0.002484228229150176, 0.002474189503118396, 0.0023066357243806124] 

Validate (0 / 10 Steps) (dice=0.00246):  67%|██████▋   | 4/6 [00:15<00:06,  3.37s/it]
Validate (0 / 10 Steps) (dice=0.00246):  83%|████████▎ | 5/6 [00:15<00:03,  3.13s/it]global step: 0 val_step: 4, 
 dice_metric.aggregate(): 0.0024649552069604397 
 dice_vals: [0.00318509666249156, 0.002484228229150176, 0.002474189503118396, 0.0023066357243806124, 0.0024649552069604397]

**Comment by ahatamiz** [comment]
Hi @yiyixuxu , thanks. I see. `np.mean(dice_vals)` in this case averages dice values (mean of 14 class) for each validation iteration.

**Comment by yiyixuxu** [comment]
Hi @ahatamiz 

Thanks.  my issue is that from my understanding, the last value in `dice_vals` is already an average for the dice values from all validation iterations (`6` in our case)  since `dice_metric` is a cumulative metric, so I don't think we should take average again with `np.mean()`?

**Comment by ahatamiz** [comment]
Hi @yiyixuxu,

I agree. We should remove it, and hence will address it in an updated version. 

Thanks

**Comment by ahatamiz** [comment]
Hi @Nic-Ma 

I will address this issue. Thanks.

**Comment by ahatamiz** [comment]
Hi @yiyixuxu 

Please check the updated tutorial.

Thanks

**Comment by yiyixuxu** [comment]
@ahatamiz  Thank you !
